client/packages/cli/commands/register/substrate.ts

```diff
@@ -32,6 +33,8 @@ const registerRelaychain = async (circuit: ApiPromise, target: ApiPromise, gatew
         gateway_genesis: await createGatewayGenesis(circuit, target),
         gateway_sys_props: createGatewaySysProps(circuit, gatewayData.registrationData.gatewaySysProps),
         allowed_side_effects: circuit.createType('Vec<AllowedSideEffect>', gatewayData.registrationData.allowedSideEffects),
+        parent_gateway_id: circuit.createType('Option<ChainId>', gatewayData.registrationData.parentChainId),
```

Comment:
How about 3 separate Gateway entries to differentiate between relays/beacons, shards/parachains, token on specific beacon/shard?

```
fn register_beacon_gateway(
    origin: OriginFor<T>,
    gateway_id: ChainId,
    codec: t3rn_abi::recode::Codec,
    gateway_abi_specs: GatewayABISpecs,
    gateway_genesis: GatewayGenesisConfig,
    token_props: TokenSysProps,
    allowed_standard_sfx_abi: Vec<Sfx4bId>,
    security_portal: u8,
    escrow_address: Option<Bytes>,
) {
}

fn register_shard_gateway(
    origin: OriginFor<T>,
    beacon_id: ChainId,
    gateway_id: ChainId,
    gateway_abi_specs: GatewayABISpecs,
    escrow_address: Option<Bytes>,
    token_props: TokenSysProps,
    allowed_standard_sfx_abi: Vec<Sfx4bId>,
) {
}

// Add a new token to the registry.
// Derives all settings from the gateway.
fn add_token(
    origin: OriginFor<T>,
    gateway_id: ChainId,
    token_base: ChainId,
    token_props: TokenSysProps,
) 
```

Reply:
security_portal: u8 is a reference to Instance of Light Client we registered in t3rn

Reply:
1) But these would all result in XDNS entries also right? Personally I would like to keep a generic extrinsic for this, but I agree that what we currently have isn't there yet.
2) What do we want to register token for? For reward payment asset? 
3) Why do you want to get rid of GatewayVendor?